### PR TITLE
Update to match the v1.10.0 release routes and parameters

### LINKED ISSUE #4671: Update Open API file for v1.10.0
This issue is about updating the [Open API file](https://github.com/meilisearch/open-api/blob/main/open-api.yaml) related to the Meilisearch release.

## How?

Open a PR on [this repository](https://github.com/meilisearch/open-api) to update the `open-api.yaml` file reflecting changes that have been implemented for the current release.

## When?

⚠️ This should be done at the end of the sprint, once all the implementations have been done.

### PR DESCRIPTION
**Pull request by irevoire**
You can find the v1.10 release notes on this page: [meilisearch@v1.10.0](https://github.com/meilisearch/meilisearch/releases/tag/v1.10.0).

Fixes https://github.com/meilisearch/meilisearch/issues/4671

### CODE REVIEW

File: open-api.yaml [hunk 1]
```diff
@@ -3430,6 +3555,88 @@ paths:
...
+          description: Ok
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/localizedAttributes'
+        '401':
+          $ref: '#/components/responses/401'
+        '404':
+          description: Not Found
+    patch:
```
Comment (ManyTheFish) [inline open-api.yaml]:
```suggestion
    put:
```

File: open-api.yaml [hunk 2]
```diff
@@ -790,6 +809,24 @@ components:
...
+            type: string
+          default: []
+          nullable: false
+        attributePatterns:
+          description: Define the attributes affected by the locales.
+          type: array
+          items:
+            type: string
+          default: []
+          nullable: false
```
Comment (ManyTheFish) [inline open-api.yaml]:
```suggestion
    localizedAttributes:
      type: array
      description: Customize the language of your index.
      items:
        type: object
        properties:
          locales:
            description: Define the languages used in the attributes.
            type: array
            items:
              type: string
            default: []
            nullable: false
          attributePatterns:
            description: Define the attributes affected by the locales.
            type: array
            items:
              type: string
            default: []
            nullable: false
```

**Review by ManyTheFish — CHANGES_REQUESTED** [review]

**Review by ManyTheFish — APPROVED** [review]
bors merge

**Comment by curquiza** [comment]
No bors here 😇

**Pull request merged**